server/controllers/recommendation/ppcs/indeterminateSentence/edit/editOffenceController.test.ts

```diff
@@ -0,0 +1,130 @@
+import { fakerEN_GB as faker } from '@faker-js/faker/'
+import { mockNext, mockReq, mockRes } from '../../../../../middleware/testutils/mockRequestUtils'
+import { ppudReferenceList, updateRecommendation } from '../../../../../data/makeDecisionApiClient'
+import { RecommendationResponseGenerator } from '../../../../../../data/recommendations/recommendationGenerator'
+import editOffenceController from './editOffenceController'
+
+jest.mock('../../../../../data/makeDecisionApiClient')
+
+const expected = {
+  basePath: `/recommendations/${faker.number.int()}/`,
+  render: {
+    path: 'pages/recommendations/ppcs/indeterminateSentence/edit/editOffence',
+  },
+}
+
+describe('Indeterminate Sentence - Edit offence controller', () => {
+  const sentenceId = faker.number.int().toString()
+  const originalOffence = faker.lorem.sentence()
+  const recommendation = RecommendationResponseGenerator.generate({
+    bookRecallToPpud: {
+      ppudSentenceId: sentenceId,
+      ppudIndeterminateSentenceData: {},
+    },
+    ppudOffender: {
+      sentences: [
+        {
+          id: sentenceId,
+          offenceDescription: originalOffence,
+        },
+      ],
+    },
+  })
+  const updatedOffenceDescription = faker.lorem.sentence()
+  const updatedOffenceDescriptionComment = faker.lorem.sentence()
+
+  describe('get', () => {
+    const req = mockReq()
+    const res = mockRes({
+      locals: {
+        recommendation,
+      },
+    })
+    const next = mockNext()
+    beforeEach(async () => {
+      ;(ppudReferenceList as jest.Mock).mockResolvedValue({ values: ['one', 'two', 'three'] })
+      await editOffenceController.get(req, res, next)
+    })
+    describe('Sets response locals correctly:', () => {
+      describe('Page Data', () => {
+        it('- is defined', async () => {
+          expect(res.locals.existingOffenceDescription).toBeDefined()
+          expect(res.locals.allOffences).toBeDefined()
+        })
+      })
+    })
+    it('Does not set any error message', async () => expect(res.locals.errorMessage).toBeUndefined())
+    it('Calls render for the expected page', async () => expect(res.render).toHaveBeenCalledWith(expected.render.path))
+    it('Executes the next function ', async () => expect(next).toHaveBeenCalled())
+  })
+
+  describe('post', () => {
+    const recommendationId = faker.number.int().toString()
+    const basePath = `/recommendations/${recommendationId}/`
+    const next = mockNext()
+
+    const validReq = mockReq({
+      params: { recommendationId },
+      body: {
+        offenceDescription: updatedOffenceDescription,
+        offenceDescriptionComment: updatedOffenceDescriptionComment,
```

Comment:
The comment is optional, is that right? If so, can we also test passing a description but no comment to make sure it doesn't error?

Reply:
Done 👍